runtime-sdk/src/crypto/signature/ed25519.rs

```diff
@@ -118,60 +118,180 @@ impl From<PublicKey> for CorePublicKey {
 
 /// A memory-backed signer for Ed25519.
 pub struct MemorySigner {
-    sk: ed25519_dalek::ExpandedSecretKey,
+    key: Key,
+}
+
+/// The original version of this signer returned the "expanded" signing key from `to_bytes`.
+/// If the contract passes in one of these, it's gets the same thing out. New invocations get the "proper" version.
```

Comment:
I propose that this is an okay thing to do because it was never documented what exactly is returned by the precompile 😐.

Reply:
It belongs on `Key`, but I can make the comment better.